src/Streamly/Internal/Data/Stream/IsStream/Top.hs

```diff
@@ -443,17 +445,29 @@ joinOuter :: MonadIO m =>
     -> SerialT m (Maybe a, Maybe b)
 joinOuter eq s1 s =
     Stream.concatM $ do
-        arr <- Array.fromStream $ fmap (,False) s
-        return $ go arr <> leftOver arr
+        arr <- Array.fromStream s
+        let len = length arr
+        arr2 <-
+            Stream.fold
+            (MA.writeN len)
+            (Stream.fromList (Prelude.replicate len False))
+        return $ go arr arr2 <> leftOver arr arr2
 
     where
 
-    leftOver =
-            fmap (\(x, _) -> (Nothing, Just x))
-                . Stream.filter (not . snd)
-                . Array.toStream
-
-    go arr = Stream.evalStateT (return False) $ do
+    leftOver arr arr2 =
+            let stream1 = IsStream.fromSerial $ Array.toStream arr
+                stream2 = IsStream.fromSerial $ fromStreamD (MA.toStreamD arr2)
```

Comment:
Why not use `MA.toStream` instead?

Reply:
I don't find toStream API :
Not in scope: ‘MA.toStream’
Perhaps you meant one of these:
  ‘MA.toStreamD’ (imported from Streamly.Internal.Data.Array.Foreign.Mut.Type),
  ‘MA.toStreamK’ (imported from Streamly.Internal.Data.Array.Foreign.Mut.Type)
Module ‘Streamly.Internal.Data.Array.Foreign.Mut.Type’ does not export ‘toStream’.typecheck

Reply:
If there is no `toStream` you can use an unfold, like `unfold MArray.read`. If we have to use `StreamD` to convert an array to stream then there is a major issue with the array API and it has to be fixed.